src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,545 @@
+         *
+         * @return name of variable
+         */
+        public String getName() {
+            return name;
+        }
+
+        /**
+         * Get the associated {@link VariableDesc#variableAst}.
+         *
+         * @return {@link VariableDesc#variableAst}
+         */
+        public DetailAST getVariableAst() {
+            return variableAst;
+        }
+
+        /**
+         * Get the associated {@link VariableDesc#scopeOfVariable}.
+         *
+         * @return {@link VariableDesc#scopeOfVariable}.
+         */
+        public DetailAST getScopeOfVariable() {
+            return scopeOfVariable;
+        }
+
+        /**
+         * Get the scope of pattern variable including line and column.
+         *
+         * @return {@link VariableDesc#scopeOfPatternVariableIncludingLineAndColumn}
+         */
+        private int[] getScopeOfPatternVariableIncludingLineAndColumn() {
+            return scopeOfPatternVariableIncludingLineAndColumn;
+        }
+
+        /**
+         * Is the variable used again or not.
+         *
+         * @return true if variable is used again else false;
+         */
+        public boolean isUsedAgain() {
+            return isUsedAgain;
+        }
+
+        /**
+         * Is the variable a pattern variable or not.
+         *
+         * @return true if variable is a pattern variable else false;
+         */
+        public boolean isPatternVariable() {
+            return isPatternVariable;
+        }
+
+        /**
+         * Register the variable as used again.
+         */
+        public void registerAsUsedAgain() {
+            this.isUsedAgain = true;
+        }
+
+        /**
+         * Register the variable as a pattern variable.
+         */
+        public void registerAsPatternVariable() {
+            this.isPatternVariable = true;
+        }
+
+        /**
+         * Set the scope of variable.
+         *
+         * @param scopeOfVariable {@link VariableDesc#scopeOfVariable}
+         */
+        public void setScopeOfVariable(DetailAST scopeOfVariable) {
+            this.scopeOfVariable = scopeOfVariable;
+        }
+
+        /**
+         * Set the scope of pattern variable including line and column.
+         *
+         * @param scopeLineStart starting line of scope
+         * @param scopeColumnStart starting column of scope
+         * @param scopeLineEnd ending line of scope
+         * @param scopeColumnEnd end column of scope
+         */
+        private void setScopeOfPatternVariableIncludingLineAndColumn(int scopeLineStart,
+                int scopeColumnStart, int scopeLineEnd, int scopeColumnEnd) {
+            scopeOfPatternVariableIncludingLineAndColumn = new int[] {
+                scopeLineStart, scopeColumnStart, scopeLineEnd, scopeColumnEnd,
+            };
+        }
+    }
+
+    /**
+     * Index values for {@link VariableDesc#scopeOfPatternVariableIncludingLineAndColumn}.
+     */
+    private static class PatternVariableScope {
```

Comment:
may be make it as enum?

Reply:
Would be inconvenient to get values from enum IMO, as we would have to make a getter method inside enum.

Reply:
you can avoid using array + indices and make it more readable using object that encapsulates this data (line start, line end, etc). Where this info is needed, it will be like
```
patternVariableScope.getLine() == identLine
```
instead of current 
```
scopeOfPatternVariableIncludingLineAndColumn[
                    PatternVariableScope.LINE_START] == identLine
```

Reply:
Will keep in mind, field not required anymore.